tests/scripts/runtest

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at the
+# Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability see http://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License (as published by the Free
+# Software Foundation) version 2.1 dated February 1999.
+
+# Set the test and run directories
+run_dir=`pwd`
+test_dir=$( cd "$( dirname "$0" )" && pwd )
+
+# Print usage information
+case $1 in
+   -h|-help)
+      cat <<EOF
+
+   $0 [-h|-help] {mfem_dir} {test1} {test2} ...
+
+   where: {mfem_dir}  is the MFEM source directory [default value: ../..]
+          {test1}...  are the sequence of tests to run
+          -h|-help    prints this usage information and exits
+
+   This script runs a sequence of tests using the code in mfem_dir (note that
+   specifying the directory is optional). For each test, the standard output and
+   standard error are saved in files test.out and test.err respectively in the
+   current directory [1]. The output file interleaves stdout and stderr, while
+   the error file is filtered by removing test-specific patterns specified in an
+   optional test.filters file in the test directory [2]. After filtering, empty
+   error files are erased, signifying that the corresponding test has passed.
+
+   [1] run directory  = $run_dir
+   [2] test directory = $test_dir
+
+   Example usage: $0 ../..
+
+EOF
+      exit
+      ;;
+esac
+
+# Setup
+if [ -d "$1" ]; then
+   mfem_dir="$1"
+   shift
+fi
+if [ "$mfem_dir" = "" ]; then
+    mfem_dir="../.."
+fi
+if ! [ -d "$mfem_dir" ]; then
+   printf "\nDirectory not found: $mfem_dir\n\n" 1>&2
+   exit 1
```

Comment:
I am not sure if this test ever failed:
if the user specifies an existing directory (tested with `-d "$1"`) then mfem_dir is set to a valid directory.
else `mfem_dir` is empty, so it’s set to "../.." which most likely exists.

Reply:
True, it is unlikely that `../..` is not a valid directory, but I think it will be fun to leave the check. I will add a comment.